chore_: improve collect log

qfrank opened the pull request:
## Summary
- Move android logs from download dir to private app data folder
- Removed unused functions
- Improved the way collecting logs on android/iOS

This PR is required before implementing log per profile, otherwise the changes introduced from status-go [PR](https://github.com/status-im/status-go/pull/6442) will be breaking change.

## Review notes
This PR depends on [PR 22275](https://github.com/status-im/status-mobile/pull/22275) and [PR 22268](https://github.com/status-im/status-mobile/pull/22268)

## Testing notes
[comment]: # (Optional)
log files should be contained when share log or report a bug:
<img width="476" alt="image" src="https://github.com/user-attachments/assets/29829fa4-c1e9-4bc9-9d5a-4ffc2cd78ef5" />


### Platforms
[comment]: # (Optional. Specify which platforms should be tested)

- Android
- iOS

### Areas that may be impacted
[comment]: # (Optional. Specify if some specific areas need to be tested, such as 1-1 chats)

#### Functional

- Share logs
- Report a bug

## Steps to test
[comment]: #  (Specify exact steps to test if there are such)

- Open Status and create new profile
- Shake device and share logs or report a bug
- See if all the files under `logs` are being zipped


[comment]: # (Can be ready or wip)
status: ready

qfrank committed:
chore_: move android logs from download dir to private app data folder

qfrank committed:
chore_: remove unused functions

qfrank committed:
chore_: rename getPublicStorageDirectory to getLogDirectory

qfrank committed:
refactor_: streamline log file handling in LogManager

Removed unused methods for retrieving specific log files and replaced them with a more dynamic approach that collects all log files from the log directory. This enhances the flexibility of log management and reduces code redundancy.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (4)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | bfe6beb9 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-22342/1/) | 2025-03-20 09:18:59 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250320-091414-bfe6be-pr22342-tests.log) |
| :heavy_check_mark: | bfe6beb9 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-22342/1/) | 2025-03-20 09:22:45 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250320-091409-bfe6be-pr22342-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250320-091409-bfe6be-pr22342-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | bfe6beb9 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-22342/1/) | 2025-03-20 09:23:39 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250320-091409-bfe6be-pr22342-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250320-091409-bfe6be-pr22342-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | bfe6beb9 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-22342/1/) | 2025-03-20 09:24:57 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/ZXkNDo) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FZXkNDo&size=400x400&errorCorrectionLevel=L)|
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 45e27702 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-22342/2/) | 2025-03-27 03:57:40 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-035141-45e277-pr22342-tests.log) |
| :heavy_check_mark: | 45e27702 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-22342/2/) | 2025-03-27 04:00:47 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-035137-45e277-pr22342-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250327-035137-45e277-pr22342-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 45e27702 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-22342/2/) | 2025-03-27 04:01:22 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-035141-45e277-pr22342-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250327-035141-45e277-pr22342-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 45e27702 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-22342/2/) | 2025-03-27 04:02:55 | ~11 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/st8xvK) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2Fst8xvK&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | ea3a15be | [#3](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-22342/3/) | 2025-03-27 07:18:40 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-071333-ea3a15-pr22342-tests.log) |
| :heavy_check_mark: | ea3a15be | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-22342/3/) | 2025-03-27 07:22:46 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-071328-ea3a15-pr22342-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250327-071328-ea3a15-pr22342-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | ea3a15be | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-22342/3/) | 2025-03-27 07:23:18 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250327-071333-ea3a15-pr22342-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250327-071333-ea3a15-pr22342-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | ea3a15be | [#3](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-22342/3/) | 2025-03-27 07:23:59 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/oRLYbZ) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FoRLYbZ&size=400x400&errorCorrectionLevel=L)|

ilmotta commented:
@qfrank This PR says it depends on https://github.com/status-im/status-mobile/pull/22268, but #22268 depends on https://github.com/status-im/status-mobile/issues/22274 (in order to not impact QAs). Therefore, it seems that we need to resolve #22274 in order to unblock #22342. Is this assessment correct?

For reference, I'm writing here the list of PRs opened about logs in status-mobile and status-go.

- https://github.com/status-im/status-mobile/pull/22342
- https://github.com/status-im/status-mobile/pull/22275
- https://github.com/status-im/status-mobile/pull/22268
- https://github.com/status-im/status-go/pull/6442
- https://github.com/status-im/status-go/pull/6405

qfrank commented:
> Therefore, it seems that we need to resolve https://github.com/status-im/status-mobile/issues/22274 in order to unblock https://github.com/status-im/status-mobile/pull/22342. Is this assessment correct?

correct @ilmotta

qfrank committed:
refactor_: remove hardcoded log file names in AccountManager and LogManager

Eliminated unused log file name constants and updated LogManager to dynamically retrieve log files from the directory, enhancing flexibility and reducing redundancy in log handling.

status-im-auto commented:
## 71% of end-end tests have passed
```
Total executed tests: 14
Failed tests: 2
Expected to fail tests: 2
Passed tests: 10
Not executed tests: 2
```
```
IDs of not executed tests: 742015,742016 
```
```
IDs of failed tests: 727231,740490 
```
```
IDs of expected to fail tests: 741840,741841 
```
<h3>Not executed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=frank/chore/improve_collect_log&PR_ID=22342&APK_NAME=22342.apk&TR_CASE_IDS=742015,742016">Rerun not executed tests</a></li></details><h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=frank/chore/improve_collect_log&PR_ID=22342&APK_NAME=22342.apk&TR_CASE_IDS=727231,740490">Rerun failed tests</a></li><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823758">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div># STEP: Adding new regular account</div><div>Device 1: Find `Button` by `accessibility id`: `add-account` </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:357: in test_wallet_add_remove_regular_account
    self.wallet_view.add_regular_account(account_name=new_account_name)
../views/wallet_view.py:245: in add_regular_account
    self.add_account_button.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `add-account` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823759">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:70: in test_wallet_balance_mainnet
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 in total
E    Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 on Optimism
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (2)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823764">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `EditBox` by `accessibility id`: `address-text-input` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:102: in test_wallet_send_collectible
    self.wallet_view.address_text_input.send_keys(self.receiver['wallet_address'])
../views/base_element.py:365: in send_keys
    self.find_element().send_keys(value)
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: EditBox by accessibility id: `address-text-input` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823765">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `collectibles-tab` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/wallet/test_collectibles.py:158: in test_wallet_collectible_send_from_expanded_info_view
    self.wallet_view.get_collectible_element('Glitch Punks').wait_for_element().click()
../views/base_element.py:120: in wait_for_element
    raise TimeoutException(
 Device `1`: `CollectibleItemElement` by` xpath`: `//*[@content-desc='collectible-list-item']//*[contains(@text,'Glitch Punks')]/../..` is not found on the screen after wait_for_element 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (10)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823753">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyFDIz4XayjfoIHNv">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjlwBnyoG8FzN0l3K">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823757">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823756">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823763">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823754">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESzJW27nn5RJZ0EOs1">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESLj8xyngbE4GaZjxh">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823752">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESVm6kiNEx3FizyMAs">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823755">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESVm6kiNEx3FizyMAs">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823761">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823762">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823760">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table></details>

Horupa-Olena commented:
@qfrank Thank you for your PR!
Everything work fine, **PR can be merged!**

Was checked:

- Shared log and Bug report feature on iOS and Android
- Presence of files in the archive
- Downloading from email
